### PR TITLE
98/ Subset expanded grid of valid values by output type

### LINKED ISSUE #98: Allow subsetting of expand_model_out_val_grid() by output type
As configs become more complex, the expanded grid of valid value combinations becomes unwieldy, hogging memory and causing validations to be very slow.

It would be better to be able to break up the validation into smaller steps. Validating one output type at a time is the easiest and most logical way to split it up. 

As such, we should introduce an additional `output_type` argument that controls which output types the expanded grid of valid values is created.

### PR DESCRIPTION
**Pull request by annakrystalli**
This PR  resolves #98 by introducing an `output_type` argument to `expand_model_out_grid()` which allows for producing output type specific subsets of the expanded grid of valid values.

It also adds a bunch of low level internal functions for extracting information from round level configs

This will be used downstream in #93 as well as form the basis for #97

### CODE REVIEW

**Comment by annakrystalli** [comment]
Apologies to the reviewer...I think I've misunderstood what rebasing was supposed to do...the last 5 commits which are somewhat polluting the PR, were through a rebase to add linting fixes made in #99 so I didn't think they'd show up here given they are already in `main`

File: tests/testthat/test-check_tbl_spl_compound_tid.R [hunk 1]
```diff
@@ -65,9 +65,10 @@ test_that("Overriding compound_taskid_set in check_tbl_spl_compound_tid works",
     )
   )
 
- # Validation providing coarser compound taskid set succeeds
+  # Validation providing coarser compound taskid set succeeds
   expect_snapshot(
     check_tbl_spl_compound_tid(tbl_coarse, round_id, file_path, hub_path,
-                               compound_taskid_set = compound_taskid_set)
+      compound_taskid_set = compound_taskid_set
+    )
```
[the remark this answers is not in the document]
Comment (elray1) [reply tests/testthat/test-check_tbl_spl_compound_tid.R:72]: I see that linting checks have passed, so I guess I was confused about what's allowed...?

**Review by elray1 — APPROVED** [review]
One question of substance if we should just always throw an error if an invalid output type is specified.  Then I had questions about indentation but it seems like linting passed so OK to just move on from those.  I've approved as good enough to merge.

**Comment by annakrystalli** [comment]
Thanks @elray1 ! Re: the linting, it's what `styler` output and as mentioned the linting passed so I think we should leave it.
> One question of substance if we should just always throw an error if an invalid output type is specified. 

I think that's reasonable.

**Pull request merged**